pandas/formats/format.py

```diff
@@ -906,8 +927,59 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if i < clevels and self.fmt.header and self.multicolumn:
+                # sum up columns to multicolumns
+                row2 = list(crow[:ilevels])
+                ncol = 1
+                coltext = ''
+
+                def append_col():
+                    # write multicolumn if needed
+                    if ncol > 1:
+                        row2.append('\\multicolumn{{{0:d}}}{{{1:s}}}{{{2:s}}}'
+                                    .format(ncol, multicolumn_format,
+                                            coltext.strip()))
+                    # don't modify where not needed
+                    else:
+                        row2.append(coltext)
+                for c in crow[ilevels:]:
+                    if c.strip():  # if next col has text, write the previous
+                        if coltext:
+                            append_col()
+                        coltext = c
+                        ncol = 1
+                    else:  # if not, add it to the previous multicolumn
+                        ncol += 1
+                if coltext:  # write last column name
+                    append_col()
+                crow = row2
+            if i >= nlevels and self.fmt.index and self.multirow:
```

Comment:
I guess, yeah. I'll add this check.
Just for curiosity, is it possible to create some clashes in a single index, e.g. via hiding parts of a MultiIndex? This thought never crossed my mind when i wrote this.